server/replication/replication_mode.go

```diff
@@ -487,10 +487,11 @@ func (m *ModeManager) estimateProgress() float32 {
 }
 
 func (m *ModeManager) checkRegionRecover(region *core.RegionInfo, startKey []byte) bool {
-	if !bytes.Equal(startKey, region.GetStartKey()) {
+	// if the region not contains the key, log it and return false
+	if bytes.Compare(startKey, region.GetStartKey()) < 0 || (len(region.GetEndKey()) != 0 && bytes.Compare(startKey, region.GetEndKey()) > 0) {
```

Comment:
I think no need to check endkey. Just `if bytes.Compare(startKey, region.GetStartKey()) < 0` should be ok.